File: corporate-reporting/src/types/UserDirectorate.ts
import { IEntity, Entity } from "./Entity";
import { IUser } from "./User";
import { IDirectorate } from "./Directorate";

export interface IUserDirectorate extends IEntity {
    UserID: number;
    DirectorateID: number;
    IsAdmin: boolean;
    IsRiskAdmin: boolean;
    User?: IUser;
    Directorate?: IDirectorate;
}

export class UserDirectorate extends Entity implements IUserDirectorate {
    public UserID = null;
    public DirectorateID = null;
    public IsAdmin = false;
    public IsRiskAdmin = false;
}

File: ORB.Core/Models/UserDirectorate.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public class UserDirectorate : EntityWithEditor
    {
        public int UserID { get; set; }
        public int DirectorateID { get; set; }
        public bool IsAdmin { get; set; }
        public bool IsRiskAdmin { get; set; }

        public virtual Directorate Directorate { get; set; }
        public virtual User User { get; set; }
    }
}
